infra/bridge/src/utils/exchange-rate.js

```diff
@@ -5,49 +5,82 @@ const request = require('superagent')
 const { getAsync, redisClient } = require('../utils/redis')
 const logger = require('../logger')
 
+const currencies = [
+  'ETH-USD',
+  'DAI-USD',
+  'KRW-USD',
+  'SGD-USD',
+  'GBP-USD',
+  'EUR-USD',
+  'JPY-USD',
+  'CNY-USD'
+]
+
 function startExchangeRatePolling() {
   // TODO: Store the markets to be polled somewhere or in ENV and start poll with that.
-  pollExchangeRate('ETH-USD')
-  pollExchangeRate('DAI-USD')
+  pollExchangeRate(currencies)
 }
 
 /**
  * Recursively polls the exchange rate in fixed intervals
  */
-async function pollExchangeRate(market) {
+async function pollExchangeRate(markets) {
   setTimeout(() => {
-    fetchExchangeRate(market).then(() => pollExchangeRate(market))
+    fetchExchangeRate(markets).then(() => pollExchangeRate(markets))
   }, process.env.EXCHANGE_RATE_POLL_INTERVAL || 30000)
 }
 
 /**
- * Fetch excahnge rate from remote URL
+ * Fetch exchange rate from coingecko
+ * Alternatively https://api.cryptonator.com/api/ticker/${market}
+ * can be used but the data payload is different and this function would need
+ * updating
+ * @param {string|Array<string>} markets - Can be single or multiple markets
  */
-async function fetchExchangeRate(market) {
-  const exchangeURL = `https://api.cryptonator.com/api/ticker/${market}`
+async function fetchExchangeRate(markets) {
+  try {
+    const exchangeURL = `https://api.coingecko.com/api/v3/exchange_rates`
+    const response = await request.get(exchangeURL)
+    if (!response.ok) {
+      throw new Error(response.error)
+    }
 
-  return new Promise(async (resolve, reject) => {
-    try {
-      const response = await request.get(exchangeURL)
-      if (!response.body.success) {
-        reject(response.body.error)
-        return
+    const rates = response.body.rates
+    if (rates) {
+      if (Array.isArray(markets)) {
+        return markets.map(market => parseAndSetRateData(market, rates))
+      } else if (typeof markets === 'string') {
+        return parseAndSetRateData(markets, rates)
+      } else {
+        throw new Error('Unexpected type for markets')
       }
-
-      resolve(response.body.ticker.price)
-    } catch (error) {
-      reject(error)
     }
-  })
-    .then(price => {
-      if (price) {
-        redisClient.set(`${market}_price`, price, 'NX')
-        logger.info(`Exchange rate for ${market} set to ${price}`)
-      }
+  } catch (e) {
+    logger.error(`Error getting ${markets} exchange rate:`, e)
+  }
+}
 
-      return price
-    })
-    .catch(e => logger.error(`Error getting ${market} exchange rate:`, e))
+function parseAndSetRateData(market, rates) {
```

Comment:
@franckc It looks like it will work with the mobile requests.

The [mobile request format](https://github.com/OriginProtocol/origin/blob/master/mobile/src/utils/price.js#L30) matches the format of the new code.  The [original fetchExchangeRate code](https://github.com/OriginProtocol/origin/commit/f825e30d9b97c4d33a0e67458c59327a6802b67e#diff-8dc11c87ee00a865639a36105ce00c22R37-R48) returned a price string and [so does my change](https://github.com/OriginProtocol/origin/blob/master/infra/bridge/src/utils/exchange-rate.js#L81-L91). Should be good! Unless I'm missing something.